### DISCUSSION TITLE
نقاش مبدئي عن طريقة التدريب

### DISCUSSION THREAD

**Discussion by mohammadmansour200** [mohammadmansour200/Tesseract-Arabic-Model#1] [General]
السلام عليكم ورحمة الله وبركاته.
https://github.com/tesseract-ocr/tessdoc/blob/main/tess5/TrainingTesseract-5.md
كما هو موضح في النص، يتوفر لدينا 3 طرق لتدريب النموذج:

>*   Fine tune. Starting with an existing trained language, train on your
    specific additional data. This may work for problems that are close to the
    existing training data, but different in some subtle way, like a
    particularly unusual font. May work with even a small amount of training
    data.
>*   Cut off the top layer (or some arbitrary number of layers) from the network
    and retrain a new top layer using the new data. If fine tuning doesn't work,
    this is most likely the next best option. Cutting off the top layer could
    still work for training a completely new language or script, if you start
    with the most similar looking script.
>*   Retrain from scratch. This is a daunting task, unless you have a very
    representative and sufficiently large training set for your problem. If not,
    you are likely to end up with an over-fitted network that does really well
    on the training data, but not on the actual data.

نحدد إن شاء الله طريقة التدريب الأمثل والأنسب لتحسين التعرف على القرآن والأرقام العربي، ومن ثم نشرع في تجميع البيانات وندربه بإذن الله.

**Comment by OmarKhalil2003** [comment]
هل ممكن تعمل كومينتي للبروجيكت؟ ديسكورد او واتساب لسهولة التواصل,  انا مهتم اشارك

**Reply by mohammadmansour200** [reply]
هل هي ضرورية؟ شخصيا لا أفضل الواتس، ولا أظن الكل يملك discord، في ظني github discussions كافية.

**Reply by mohammadmansour200** [reply]
لو ضرورية، نستعمل تيليغرام أنسب

**Comment by AhmedEssam24933** [comment]
طيب بالنسبة لموضوع ال word segmentation ازاي المودل هيقدر يفصل كلمات القرآن عن بعض وهي مختلفة عن شكل الجمل اللي اتدرب عليها؟ عشان الخط العثماني الكلمات بتكون أقرب لبعض من الخط العادي وأحياناً آخر حرف في الكلمة بيكون لازق في أول حرف في الكلمة اللي بعدها ، هو لو معرفش يفصل الكلمات كويس مش هيعرف يفصل كل كلمة لحروف ، تاني حاجة القرآن فيه حروف مختلفة في الشكل زي حرف الألف ساعات بيتكتب زي رقم 1 لو في نص الكلام وحروف تانية بتكون فوق بعض الكلمات زي ج اللي بتستخدمها عشان تعرف إنك المفروض تقف هنا ، بالإضافة للتشكيل برضو ، كل ده بيخلي الموضوع صعب ويمكن بيخلي المودل يعتبرها كإنها لغة تانية أصلاً ومختلفة بشكل كبير عن شكل الداتا اللي اتدرب عليها

**Reply by mohammadmansour200** [reply]
>  ازاي المودل هيقدر يفصل كلمات القرآن عن بعض وهي مختلفة عن شكل الجمل اللي اتدرب عليها؟

بسيطة جدا إن شاء الله، الخط العثماني سهل جدا نحمله كـfont من على هذا الرابط:
https://arabicfonts.net/fonts/kfgqpc-uthmanic-script-hafs-regular
الـfont رسمي وتابع لمجمع الملك فهد لطباعة المصحف.

وكما أن صاحب هذا [المنشور](https://medium.com/@riasharma.indian/training-tesseract-ocr-with-custom-data-and-font-cf9e94429b4a) تمكن من تدريب خط بهذا الشكل لدعمه في الإنجليزي:
![Blackadder ITC](https://github.com/user-attachments/assets/34a2699b-c39f-461b-b671-8d90cea81b2e)

فلا إشكال أبدا في تدريب tesseract لدعم الخط العثماني


---

> تاني حاجة القرآن فيه حروف مختلفة في الشكل زي حرف الألف ساعات بيتكتب زي رقم 1 لو في نص الكلام وحروف تانية بتكون فوق بعض الكلمات زي ج اللي بتستخدمها عشان تعرف إنك المفروض تقف هنا 

لا حاجة لدعم حروف الوقف، بإمكاننا تجاهلها، بمعنى لا يتم إرجاعها مع الكلام كنتيجة. وألقِ نظرة على المنشور السابق، فهو مبسط وموضح وسيجيب عن أسئلتك.

تعديل: الرابط هذا مفيد، ألق عليه نظرة: https://arcruz0.github.io/posts/finetuning-tess/index.html

الموضوع بسيط وسهل إن شاء الله.

**Comment by sniper0110** [comment]
السلام عليكم 
ممكن تشارك معنا الdata لي عندك؟
ستعطينا فكرة افضل عما يمكن فعله حاليا.

**Reply by mohammadmansour200** [reply]
وعليكم السلام ورحمة الله وبركاته
 أولا: القرآن كصورة لآيات متنوعة والنص الموازي له متوفرة، ثانيا، أملك مئات الكتب الشرعية كنص

**Reply by sniper0110** [reply]
ممكن تشارك الdata صورة لآيات متنوعة والنص الموازي له?

**Reply by mohammadmansour200** [reply]
ها هي: https://github.com/mohammadmansour200/Tesseract-Arabic-Model/raw/refs/heads/main/quran-ayahs.zip
لكن للأسف ما زال بحاجة لتحويله لملف .box لأن لولاه لن يعرف tesseract مكان الأحرف

هذه تحتاج متطوعين، كل شخص له مقدار معين، إلى أن ننهي ال5000~ صورة

**Reply by mohammadmansour200** [reply]
عندي فكرة أفضل، أعمل عليها إن شاء الله وأوافيك بخبرها أفضل من التطوع وما التطوع.

**Reply by mohammadmansour200** [reply]
السلام عليكم ورحمة الله وبركاته، حاولت أدرب على سورة الفاتحة وأول وجهين في البقرة كتجربة، وجربت على صورة تحتوى على آية واحدة من ضمن ما تم تدريبه عليه تماما، للأسف النتيجة سيئة جدا، بدأت أقتنع بكلام الأخ عن صعوبة الموضوع: https://github.com/mohammadmansour200/Tesseract-Arabic-Model/discussions/1#discussioncomment-11690040

**Reply by mohammadmansour200** [reply]
أنشأت الريبو هذا فيه كل شيء استعملته للتدريب ورتبته ورفعته: https://github.com/mohammadmansour200/tesseract_finetune
هذا الجزء يحتوى على ملفات box: https://github.com/mohammadmansour200/tesseract_finetune/tree/main/tesstrain/data/ara-ground-truth

**Comment by farsabdraouf** [comment]
https://huggingface.co/spaces/MohamedRashad/Arabic-Nougat
